sdk/servicebus/Microsoft.Azure.ServiceBus/src/ServiceBusConnectionStringBuilder.cs

```diff
@@ -358,11 +370,33 @@ void ParseConnectionString(string connectionString)
                         throw Fx.Exception.Argument(nameof(connectionString), $"The {OperationTimeoutConfigName} ({value}) must be smaller than one hour.");
                     }
                 }
+                else if (key.Equals(AuthenticationConfigName, StringComparison.OrdinalIgnoreCase))
+                {
+                    this.Authentication = value;
+                }
                 else
                 {
                     ConnectionStringProperties[key] = value;
                 }
             }
+            validate();
+        }
+
+        void validate()
+        {
```

Comment:
nit: Instead of calling during ToString(), you could just do validation within the setter of each of the property..

Reply:
Given that you have validation within the setter, you don't need this method anymore.

---
In reply to: [286699955](https://github.com/Azure/azure-sdk-for-net/pull/6393#discussion_r286699955) [](ancestors = 286699955)